lib/shared.bash

```diff
@@ -164,10 +164,29 @@ function build_image_override_file_with_version() {
     service_name=$1
     image_name=$2
     target=$3
-    cache_from_amt=$4
-    shift 4
+    shift 3
+
+    # load cache_from array
+    cache_from=()
+    if (( $# > 0 )); then
+      cache_from_amt=$1
+      shift
+      while (( cache_from_amt-- > 0 )); do
+        cache_from+=( "$1" ); shift
+      done
+    fi
+
+    # load labels array
+    labels=()
+    if (( $# > 0 )); then
+      labels_amt=$1
+      shift
+      while (( labels_amt-- > 0 )); do
+        labels+=( "$1" ); shift
+      done
```

Comment:
same here:
for style consistency with the rest of the code I would re-write this to the equivalent (but untested):


```suggestion
    labels_amt="${1:-0}"

    if [[ "${labels_amt}" -gt 0 ]] ; then
      shift; # remove the value
      for amt in $(seq 1 "$labels_amt"); do
        labels+=( "$1" ); shift
      done
    fi
```